radio_beam/commonbeam.py

```diff
@@ -0,0 +1,522 @@
+            transform_ellipse(small_major, small_minor, pa_diff, p, q)
+
+        # The transformed minor axis is major_comb, as defined in CASA
+        trans_minor_sc = major_comb
+
+        # Return beam to the original coordinates, still rotated with
+        # the major along the x axis
+        trans_major_unsc, trans_minor_unsc, trans_pa_unsc = \
+            transform_ellipse(trans_major_sc, trans_minor_sc,
+                              trans_pa_sc, 1 / p, 1 / q)
+
+        # Lastly, rotate the PA to the enclosing ellipse
+        trans_major = trans_major_unsc.to(u.arcsec)
+        trans_minor = trans_minor_unsc.to(u.arcsec)
+        trans_pa = trans_pa_unsc + large_beam.pa
+
+        # The minor axis becomes an issue when checking against the smaller
+        # beam from deconvolution. Adding a tiny fraction makes the deconvolved
+        # beam JUST larger than zero (~1e-7).
+        epsilon = 100 * np.finfo(trans_major.dtype).eps * trans_major.unit
+        trans_beam = Beam(major=trans_major + epsilon,
+                          minor=trans_minor + epsilon,
+                          pa=trans_pa)
+
+        # Ensure this beam can now be deconvolved
+        deconv_large_beam = \
+            trans_beam.deconvolve(large_beam,
+                                  failure_returns_pointlike=True)
+        deconv_prob_beam = \
+            trans_beam.deconvolve(small_beam,
+                                  failure_returns_pointlike=True)
+        if not deconv_large_beam.isfinite or not deconv_prob_beam.isfinite:
+            raise BeamError("Failed to find common beam that both beams can "
+                            "be deconvolved by.")
+
+        # Taken from CASA implementation, but by adding epsilon, this shouldn't
+        # be needed
+
+        # Scale the enclosing ellipse by a small factor until it does.
+        # can_deconv = False
+        # num = 0
+        # while not can_deconv:
+
+        #     deconv_large_beam = \
+        #         trans_beam.deconvolve(large_beam,
+        #                               failure_returns_pointlike=True)
+        #     deconv_prob_beam = \
+        #         trans_beam.deconvolve(small_beam,
+        #                               failure_returns_pointlike=True)
+
+        #     if (not deconv_large_beam.isfinite or not deconv_prob_beam.isfinite):
+
+        #         scale_factor = 1.001
+        #         trans_beam = Beam(major=trans_major * scale_factor,
+        #                           minor=trans_minor * scale_factor,
+        #                           pa=trans_pa)
+        #     else:
+        #         can_deconv = True
+
+        #     if num == 10:
+        #         break
+
+        #     num += 1
+
+        common_beam = trans_beam
+        return common_beam
+
+
+def boundingcircle(bmaj, bmin, bpa):
+    thisone = np.argmax(bmaj)
+    return bmaj[thisone], bmaj[thisone], bpa[thisone]
+
+
+def PtoA(bmaj, bmin, bpa):
```

Comment:
I get that it's "something" to a rotation + stretching matrix... what is `P`?  Maybe add a docstring comment along the lines of: `"""Converts a set of beam parameters to a rotation matrix"""` or similar?